src/test/regress/expected/set_operations.out

```diff
@@ -658,22 +668,24 @@ DEBUG:  Plan 137 query after replacing subqueries and CTEs: SELECT x, y FROM rec
 
 -- correlated subquery with union in WHERE clause
 SELECT * FROM test a WHERE x IN (SELECT x FROM test b UNION SELECT y FROM test c WHERE a.x = c.x) ORDER BY 1,2;
-DEBUG:  generating subplan 141_1 for subquery SELECT x FROM recursive_union.test b
+DEBUG:  generating subplan 144_1 for subquery SELECT x FROM recursive_union.test b
+DEBUG:  skipping recursive planning for the subquery since it contains references to outer queries
+DEBUG:  skipping recursive planning for the subquery since it contains references to outer queries
```

Comment:
a thought to consider for the future is that when there are correlated subqueries, we may want to prefer them as an anchor.  

(e.g. for situations like `<subquery partitioned by A> join <subquery partitioned by B> where <correlated subquery partitioned by B>`). We'd fail to plan this query because we'd fail to recursively plan the last subquery, but we'd succeed if it was the anchor.)